src/lib/add-on/clients/inst_add-on.rb

```diff
@@ -0,0 +1,206 @@
+        # to the caller
+        @ret = ProductControl.RunFrom(
+          Ops.add(ProductControl.CurrentStep, 1),
+          true
+        )
+        @ret = :finish if @ret == :next
+      end
+
+      @ret
+
+      # EOF
+    end
+
+    # Ask to user to configure network for installing remote addons.
+    # User is aske when there is a remote add-on found and the network
+    # is not running yet.
+    #
+    # @param addon_urls list of URLs
+    # @return symbol user input result (`next, `back or `abort)
+    def NetworkSetupForAddons(addon_urls)
+      addon_urls = deep_copy(addon_urls)
+      # protocols locally available (no network needed)
+      local_protocols = ["cd", "dvd", "hd"]
+
+      # is this CD/DVD/HDD installation?
+      if Builtins.contains(
+          local_protocols,
+          Builtins.tolower(Linuxrc.InstallInf("InstMode"))
+        )
+        # is there any remote addon requiring network setup?
+        network_needed = false
+        Builtins.foreach(addon_urls) do |url|
+          # is it a remote protocol?
+          if !Builtins.contains(
+              local_protocols,
+              Builtins.tolower(Ops.get_string(URL.Parse(url), "scheme", ""))
+            )
+            network_needed = true
+            raise Break
+          end
+        end
+
+
+        if network_needed
+          # check and setup network
+          ret = Convert.to_symbol(WFM.CallFunction("inst_network_check", []))
+          Builtins.y2milestone("inst_network_check ret: %1", ret)
+
+          return ret if Builtins.contains([:back, :abort], ret)
+        end
+      end
+
+      :next
+    end
+
+    # get the URLs of the all present add-ons
+    # @return [Array<String>] list of URLs (empty list if no add-on defined)
+    def current_addons
+      AddOnProduct.add_on_products.map do |addon|
+        Pkg.SourceURL(addon["media"])
+      end
+    end
+
+    # URL schemas to be considered as CD/DVD
+    DVD_SCHEMES = ["cd", "dvd"].freeze
+
+    # Determine whether the URL points to the installation CD/DVD
+    #
+    # @param url [String] Add-on URL
+    # @return [Boolean] true if it points to the installation CD/DVD
+    def instsys_dvd?(url)
+      scheme = URL.Parse(url)["scheme"]
+      return false unless DVD_SCHEMES.include?(scheme)
+
+      addon_device = device_from_url(url)
+      instsys_device && addon_device && addon_device == instsys_device
+    end
+
+    # instsys device
+    #
+    # @return [String] instsys device; nil if the device could not be
+    #   determined (for instance, when using a network based installation media)
+    def instsys_device
+      @instsys_device ||= device_from_url(InstURL.installInf2Url)
+    end
+
+    # Real device from an URL
+    #
+    # @param url [String] URL
+    # @return [String,nil] Path to the device; nil if no device is specified
+    #   or the device does not exist.
+    def device_from_url(url)
+      parsed = URL.Parse(url)
```

Comment:
NP: I'd prefer the native Ruby `URI` class here, but I'm fine with `Yast::URL`...

Reply:
Yes, me too. But just for consistency...